### PR TITLE
Changes to restructure code branch

### PR DESCRIPTION
**Pull request by yavvorn**
Changes that were made:

- main - using dictreader instead of csv reader
- added a func to calculate slice size
- added a line of code to set the background to grey

spending.csv - added title for each column

test_main - added simple unit tests

toml - started to use poetry as a dependency manager

the other 3 files are automatically generated

### CODE REVIEW

File: install-pyenv-win.ps1 [hunk 1]
```diff
@@ -0,0 +1,151 @@
+<#
+    .SYNOPSIS
```
Comment (slavovthinks) [inline install-pyenv-win.ps1:2]: Remove this file from the project. Whoever uses this project is responsible to install pyenv by himself if needed

File: tests/test_main.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        self.assertEqual(autopct_format(25), '£104.28\n(25.00%)')
+        self.assertEqual(autopct_format(50), '£208.57\n(50.00%)')
+        self.assertEqual(autopct_format(75), '£312.86\n(75.00%)')
+
+    def test_valid_category_totals(self):
+        self.assertEqual(overall_usage.get('Food', 0), 163.0)
+        self.assertEqual(overall_usage.get('Entertainment', 0), 75.0)
+
+    def test_calculate_totals(self):
+        totals = {}
```
Comment (slavovthinks) [inline tests/test_main.py:29]: this test doesnt test anything in main. It tests if the code written inside the test itself works properly.

File: tests/test_main.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import csv
+import unittest
+from spending_tracker.main import autopct_format, overall_usage
+
+
+class TestCSVData(unittest.TestCase):
```
Comment (slavovthinks) [inline tests/test_main.py:6]: Tests cant execute at least on my machine

File: tests/test_main.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                        "14,Cash\n" \
+                        "15.14,Bills\n"
+
+    def test_autopct_format(self):
+        self.assertEqual(autopct_format(25), '£104.28\n(25.00%)')
+        self.assertEqual(autopct_format(50), '£208.57\n(50.00%)')
+        self.assertEqual(autopct_format(75), '£312.86\n(75.00%)')
+
+    def test_valid_category_totals(self):
+        self.assertEqual(overall_usage.get('Food', 0), 163.0)
```
Comment (slavovthinks) [inline tests/test_main.py:25]: how is that expected to work?

File: spending.csv [hunk 5]
```diff
@@ -1,4 +1,4 @@
-amount,category
+Amount,Category
```
Comment (slavovthinks) [inline spending.csv:1]: why capital letters?

**Review by slavovthinks — CHANGES_REQUESTED** [review]
There are no installation instructions added to the README, there are no instructions on how to run the tests added to the README.
The `main.py` code is hard to test. 
It should be split in functions, also the classical `if __name__ == "__main__":` should be used. And all the code should be within a main function which gets called.
https://www.theserverside.com/tip/What-does-the-Python-if-name-equals-main-construct-do

**Comment by yavvorn** [comment]
Have made the following changes alongside a fair bit of errors when it comes to commiting/pushing. 
Readme has been updated accordingly. 
main.py - now has appropriate functions with each action. 
test_main.py - now has appropriate unittests (The first two were created with the help of chatgpt).
Changed in the csv file the titles to lower letter.

**Review by slavovthinks — CHANGES_REQUESTED** [review]

**Pull request merged**